City-of-Helsinki/helsinki-paatokset
# UHF-9677: Add descriptions to non-public attachments

# [UHF-9677](https://helsinkisolutionoffice.atlassian.net/browse/UHF-9677)

## What was done
Adds a description to non-public attachments.

## How to install

* Make sure your instance is up and running on correct branch.
  * `git checkout UHF-9677-missing-link-description`
  * If you don't have the test of production databases locally, follow these instructions: https://helsinkisolutionoffice.atlassian.net/wiki/spaces/PP/pages/6151897290/Lokaali+kehitysymp+rist#Kehitysymp%C3%A4rist%C3%B6n-pystytt%C3%A4minen%3A 
    * You don't need to pull any test data, only the entities described below 
* Run `make drush-updb drush-cim` 
* Get test data:
  * `drush ap:update cases hel-2024-001754 -v; drush ap:update decisions 8c59e987-fce2-c371-856a-8fa5a7900000 -v`
* Update translations: `drush locale:check && drush locale:update; drush cr` 

## How to test
* Go to https://helsinki-paatokset.docker.so/fi/asia/hel-2024-001754/8c59e987-fce2-c371-856a-8fa5a7900000
  * [x] The attachments marked with an X should have a label below the title and it should be translated correctly:
    * FI: Liitettä ei julkaista internetissä.
    * SV: Bilagan publiceras inte på internet.
    * EN: The attachment will not be published on the internet.  
* Go to https://helsinki-paatokset.docker.so/fi/asia/hel-2024-001754 and click edit (you should be editing the case node, not the decision).
  * Scroll down to the "Publicity Class" field and change it to "Salassa pidettävä", then save the node.
  * [x] There should now be a red tag marked "Salassa pidettävä" under the main heading.
  * [x] Check that the tag is translated to finnish, swedish and english.

## Continuous documentation

## Translations


[UHF-9677]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-9677?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
